File: reynolds-delicious-restaurant/src/com/reynolds/decious/restaurant/entities/Menu.java
package com.reynolds.decious.restaurant.entities;

import java.util.List;
import java.util.Objects;

public class Menu {
    private Long id;
    private String name;
    private List<MenuItem> finalProducts;

    public Menu() {
    }

    public Menu(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<MenuItem> getFinalProducts() {
        return finalProducts;
    }

    public void setFinalProducts(List<MenuItem> finalProducts) {
        this.finalProducts = finalProducts;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Menu menu = (Menu) o;
        return Objects.equals(id, menu.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: jmillord-accelerated-restaurant/src/entities/Menu.ts
import { FinalProduct } from './FinalProduct'
import { MenuItem } from './MenuItem'

import { Md5 } from 'ts-md5/dist/md5'


export class Menu {
  private id: string;
  private name: string;
  private finalProducts: Array<MenuItem>

  constructor() { }

  public getId(): string {
    return this.id;
  }

  public setId(id: string): void {
    this.id = id;
  }

  public getName(): string {
    return this.name;
  }

  public setName(name: string): void {
    this.name = name;
  }

  public getFinalProducts(): Array<MenuItem> {
    return this.finalProducts;
  }

  public setFinalProducts(finalProducts: Array<MenuItem>): void {
    this.finalProducts = finalProducts;
  }

  public hashCode() {
    return Md5.hashStr(this.id)
  }



}